<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>CoreEulerAnglesEditorWidget</class>
 <widget class="QWidget" name="CoreEulerAnglesEditorWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>172</width>
    <height>122</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_2">
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_2">
     <item>
      <layout class="QFormLayout" name="formLayout_2">
       <item row="0" column="0">
        <widget class="QLabel" name="euler1Label">
         <property name="text">
          <string>X:</string>
         </property>
         <property name="buddy">
          <cstring>euler1DoubleSpinBox</cstring>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QDoubleSpinBox" name="euler1DoubleSpinBox">
         <property name="suffix">
          <string>°</string>
         </property>
         <property name="decimals">
          <number>2</number>
         </property>
         <property name="minimum">
          <double>-360.000000000000000</double>
         </property>
         <property name="maximum">
          <double>360.000000000000000</double>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QLabel" name="euler2Label">
         <property name="text">
          <string>Y:</string>
         </property>
         <property name="buddy">
          <cstring>euler2DoubleSpinBox</cstring>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QDoubleSpinBox" name="euler2DoubleSpinBox">
         <property name="suffix">
          <string>°</string>
         </property>
         <property name="minimum">
          <double>-360.000000000000000</double>
         </property>
         <property name="maximum">
          <double>360.000000000000000</double>
         </property>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QLabel" name="euler3Label">
         <property name="text">
          <string>Z:</string>
         </property>
         <property name="buddy">
          <cstring>euler3DoubleSpinBox</cstring>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <widget class="QDoubleSpinBox" name="euler3DoubleSpinBox">
         <property name="suffix">
          <string>°</string>
         </property>
         <property name="minimum">
          <double>-360.000000000000000</double>
         </property>
         <property name="maximum">
          <double>360.000000000000000</double>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <widget class="QRadioButton" name="degreesRadioButton">
         <property name="text">
          <string>Degrees</string>
         </property>
         <property name="checked">
          <bool>true</bool>
         </property>
         <attribute name="buttonGroup">
          <string notr="true">angleRepresentationButtonGroup</string>
         </attribute>
        </widget>
       </item>
       <item>
        <widget class="QRadioButton" name="radiansRadioButton">
         <property name="text">
          <string>Radians</string>
         </property>
         <attribute name="buttonGroup">
          <string notr="true">angleRepresentationButtonGroup</string>
         </attribute>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <widget class="QLabel" name="eulerAnglesConventionLabel">
       <property name="text">
        <string>Convention:</string>
       </property>
       <property name="buddy">
        <cstring>eulerAnglesConventionComboBox</cstring>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QComboBox" name="eulerAnglesConventionComboBox"/>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>euler1DoubleSpinBox</sender>
   <signal>valueChanged(double)</signal>
   <receiver>CoreEulerAnglesEditorWidget</receiver>
   <slot>updateValue()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>58</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>85</x>
     <y>60</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>euler2DoubleSpinBox</sender>
   <signal>valueChanged(double)</signal>
   <receiver>CoreEulerAnglesEditorWidget</receiver>
   <slot>updateValue()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>58</x>
     <y>46</y>
    </hint>
    <hint type="destinationlabel">
     <x>85</x>
     <y>60</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>euler3DoubleSpinBox</sender>
   <signal>valueChanged(double)</signal>
   <receiver>CoreEulerAnglesEditorWidget</receiver>
   <slot>updateValue()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>58</x>
     <y>72</y>
    </hint>
    <hint type="destinationlabel">
     <x>85</x>
     <y>60</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>angleRepresentationButtonGroup</sender>
   <signal>buttonClicked(int)</signal>
   <receiver>CoreEulerAnglesEditorWidget</receiver>
   <slot>changeAngleRepresentation(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>85</x>
     <y>60</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>eulerAnglesConventionComboBox</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>CoreEulerAnglesEditorWidget</receiver>
   <slot>changeConvention(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>125</x>
     <y>101</y>
    </hint>
    <hint type="destinationlabel">
     <x>85</x>
     <y>60</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <signal>valueChanged(osg::Quat)</signal>
  <slot>setValue(osg::Quat)</slot>
  <slot>setValue(osg::Vec3,bool,int)</slot>
  <slot>updateValue()</slot>
  <slot>changeAngleRepresentation(int)</slot>
  <slot>changeConvention(int)</slot>
 </slots>
 <buttongroups>
  <buttongroup name="angleRepresentationButtonGroup"/>
 </buttongroups>
</ui>
